client/my-sites/checkout/composite-checkout/components/prepurchase-notices/index.tsx

```diff
@@ -49,45 +51,73 @@ const PrePurchaseNotices = () => {
 		return getSitePlan( state, siteId );
 	} );
 
-	const getMatchingProducts = ( items, planSlug, isCartItem ) => {
-		const planFeatures = getAllFeaturesForPlan( planSlug );
+	const getMatchingProducts = useCallback( ( items: ResponseCartProduct[], planSlug: string ) => {
+		const planFeatures = getAllFeaturesForPlan( planSlug ) as ReadonlyArray< string >;
 
 		return items.filter( ( item ) => {
-			const productSlug = isCartItem ? item.product_slug : item.productSlug;
-			//some products have had many variations over the years so we need to check an abstraction for backup, scan and Akismet.
-			return (
-				planFeatures.includes( productSlug ) ||
-				planHasSuperiorFeature( planSlug, productSlug ) ||
-				isJetpackBackupSlug( productSlug ) ||
-				isJetpackScanSlug( productSlug ) ||
-				isJetpackAntiSpamSlug( productSlug )
-			);
+			const productSlug = item.product_slug;
+
+			const productFoundInPlanFeatures =
+				planFeatures.includes( productSlug ) || planHasSuperiorFeature( planSlug, productSlug );
+
+			if ( productFoundInPlanFeatures ) {
+				return true;
+			}
+
+			if ( isJetpackPlanSlug( planSlug ) ) {
+				if ( isJetpackBackupSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_BACKUP_PRODUCTS );
+				}
+				if ( isJetpackScanSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_SCAN_PRODUCTS );
+				}
+				if ( isJetpackAntiSpamSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_ANTI_SPAM_PRODUCTS );
+				}
+				if ( isJetpackVideoPressSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_VIDEOPRESS_PRODUCTS );
+				}
+				if ( isJetpackBoostSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_BOOST_PRODUCTS );
+				}
+				if ( isJetpackSocialSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_SOCIAL_PRODUCTS );
+				}
+				if ( isJetpackSearchSlug( productSlug ) ) {
+					return planFeaturesIncludesProducts( planFeatures, JETPACK_SEARCH_PRODUCTS );
+				}
+			}
+
+			return false;
 		} );
-	};
+	}, [] );
 
 	const cartProductThatOverlapsSitePlan = useMemo( () => {
 		const planSlugOnSite = currentSitePlan?.product_slug;
 
+		// Bypass the notice if the site doesn't have a plan or the plan is Jetpack Free
 		if ( ! planSlugOnSite || planSlugOnSite === 'jetpack_free' ) {
 			return null;
 		}
 
-		const matchingProducts = getMatchingProducts( responseCart.products, planSlugOnSite, true );
+		const matchingProducts = getMatchingProducts( responseCart.products, planSlugOnSite );
```

Comment:
Why do we remove the `true` here? I see that it is for `isCartItem`

Reply:
In the `getMatchingProducts` function definition I removed the `isCartItem` argument because I felt it wasn't serving any purpose in this component. Since we're passing in an array of ResponseCartProduct(s) (`items: ResponseCartProduct[]`), it's assumed that each one is a cart item (`isCartItem`).  And this is not an exported function that is used by anything else that would possibly be passing in non-responseCartItems (such as SelectorProducts or anything) so therefore there's no need to distinguish the items _are_ or _are not_ cart items because we know they _are_.  Does that make sense?